tracer/test/Datadog.Trace.Security.IntegrationTests/AspNetCore5AsmInitialization.cs

```diff
@@ -4,36 +4,81 @@
+            : base(fixture, outputHelper, enableSecurity: false, ruleset: "wrong-tags-name-rule-set.json", testName: nameof(AspNetCore5AsmInitializationSecurityDisabledWithBadRuleset))
+        {
+        }
+    }
+
+    public class AspNetCore5AsmInitializationSecurityEnabled : AspNetCore5AsmInitialization
+    {
+        public AspNetCore5AsmInitializationSecurityEnabled(AspNetCoreTestFixture fixture, ITestOutputHelper outputHelper)
+            : base(fixture, outputHelper, enableSecurity: true, ruleset: null, testName: nameof(AspNetCore5AsmInitializationSecurityEnabled))
+        {
+        }
+    }
+
+    public class AspNetCore5AsmInitializationSecurityEnabledWithBadRuleset : AspNetCore5AsmInitialization
+    {
+        public AspNetCore5AsmInitializationSecurityEnabledWithBadRuleset(AspNetCoreTestFixture fixture, ITestOutputHelper outputHelper)
+            : base(fixture, outputHelper, enableSecurity: true, ruleset: "wrong-tags-name-rule-set.json", testName: nameof(AspNetCore5AsmInitializationSecurityEnabledWithBadRuleset))
+        {
+        }
+    }
+
+    public abstract class AspNetCore5AsmInitialization : AspNetBase, IClassFixture<AspNetCoreTestFixture>
+    {
+        public AspNetCore5AsmInitialization(AspNetCoreTestFixture fixture, ITestOutputHelper outputHelper, bool enableSecurity, string ruleset, string testName)
+            : base("AspNetCore5", outputHelper, "/shutdown", testName: testName)
+        {
+            Fixture = fixture;
+            Fixture.SetOutput(outputHelper);
+            EnableSecurity = enableSecurity;
+            RuleSet = ruleset;
+        }
+
+        protected AspNetCoreTestFixture Fixture { get; }
+
+        protected bool EnableSecurity { get; }
+
+        protected string RuleSet { get; }
+
+        public override void Dispose()
+        {
+            base.Dispose();
+            Fixture.SetOutput(null);
+        }
 
-        [SkippableTheory]
-        [InlineData(true, HttpStatusCode.OK)]
-        [InlineData(false, HttpStatusCode.OK)]
-        [InlineData(true, HttpStatusCode.OK, "wrong-tags-name-rule-set.json")]
-        [InlineData(false, HttpStatusCode.OK, "wrong-tags-name-rule-set.json")]
+        [SkippableFact]
         [Trait("RunOnWindows", "True")]
-        public async Task TestSecurityInitialization(bool enableSecurity, HttpStatusCode expectedStatusCode, string ruleset = null)
+        public async Task TestSecurityInitialization()
         {
             var url = "/Health/?[$slice]=value";
-            var agent = await RunOnSelfHosted(enableSecurity, externalRulesFile: ruleset);
-            var settings = VerifyHelper.GetSpanVerifierSettings(enableSecurity, (int)expectedStatusCode, ruleset);
-            await TestAppSecRequestWithVerifyAsync(agent, url, null, 1, 1, settings, testInit: true);
+            await Fixture.TryStartApp(this, enableSecurity: EnableSecurity, externalRulesFile: RuleSet, sendHealthCheck: false);
```

Comment:
Minor thing, and it's what was happening before I believe, but I'd argue `enableSecurity` and `exeternalRulesFile` shouldn't be parameters to this method - they can be environment variables simply set on `this.EnvironmentHelper`.

Reply:
Yeah I agree. I was thinking of other ways to refactor `TestHelper` / `EnvironmentHelper` so I'm inclined to postpone this change for a later PR